features/steps/thens/relations.py

```diff
@@ -133,41 +133,44 @@ def step_impl(context, related, relating, other_entity, condition):
     misc.handle_errors(context, errors)
 
 
+@then('The IfcPropertySet Name attribute value must use predefined values according to the {table} table')
 @then('The IfcPropertySet must be assigned according to the property set definitions table {table}')
 def step_impl(context, table):
     if getattr(context, 'applicable', True):
         errors = []
         tbl_path = system.get_abs_path(f"resources/property_set_definitions/{table}")
         tbl = system.get_csv(tbl_path, return_type='dict')
 
-        property_set_definitons = {d['Name']: d['ApplicableClasses'] for d in tbl}
-        property_set_definitons = {key: value.split(';') for key, value in property_set_definitons.items()}
+        property_set_definitons = {}
+        for d in tbl:
+            if property_set_definitons.get(d['Name']):
+                property_set_definitons[d['Name']] = property_set_definitons[d['Name']] + [d['ApplicableClasses']]
+            else:
+                property_set_definitons[d['Name']] = [d['ApplicableClasses']]
 
         for inst in context.instances:
             if isinstance(inst, (tuple, list)):
                 inst = inst[0]
 
             name = getattr(inst, 'Name', 'Attribute not found')
 
-            if name not in property_set_definitons.keys():
-                errors.append(err.InvalidValueError(False, inst, 'Name', name))  # A custom Pset_ prefixed attribute, e.g. Pset_Mywall
-                continue
-
-            try:
-                relation = inst.PropertyDefinitionOf[0]  # IFC2x3
-            except AttributeError:
-                relation = inst.DefinesOccurrence[0]  # IFC4x3
-            except IndexError:  # IfcPropertySet not assigned, rule not further checked
-                continue
+            if 'IfcPropertySet Name attribute value must use predefined values according' in context.step.name:
```

Comment:
@aothms, @Ghesselink 

Perhaps that's another way to handle multiple decorators, that is to check the context.step.name. 
It's a bit of a brute force solution, nothing too elegant, but combined with using some additional functions for readability it might work as a short-term solution.